src/libs/PDCurses/sdl2_queue/pdcdisp.cpp

```diff
@@ -563,18 +563,17 @@ void PDC_doupdate(void)
 
 void PDC_pump_and_peep(void)
 {
-    SDL_Event event;
-
-    if (SDL_PollEvent(&event))
+    if (!pdc_event_queue.empty())
     {
+        const auto event = pdc_event_queue.front();
```

Comment:
The SDL is a pretty big struct, which `front()` is returning as a copy.
suggest `const auto &` to get a reference.

Reply:
`sizeof(SDL_Event)` is only 56 bytes, but I can make this change here.

Reply:
actually, it looks like `front` already returns a reference, https://en.cppreference.com/w/cpp/container/queue/front

Reply:
When a reference is returned, it can be assigned into a full type or a reference.

`auto` on it's own is always the "full type" (and won't be a reference or pointer to the returned variable).

```c++
#include <stdio.h>

int a = 1;

int & get_ref_of_a() {
    return a;
}

int main()
{
    auto maybe_ref_of_a = get_ref_of_a();
    maybe_ref_of_a++;

    // // if it's really a ref, then 'a' will be incremented
    printf("maybe_ref_of_a = %d, a = %d\n", maybe_ref_of_a, a);

    auto & ref_of_a = get_ref_of_a();
    ref_of_a++;
    printf("ref_of_a = %d, a = %d\n", ref_of_a, a);

    return 0;
}
```

result:

```
maybe_ref_of_a = 2, a = 1
ref_of_a = 2, a = 2
```

Reply:
> `sizeof(SDL_Event)` is only 56 bytes, but I can make this change here.

Thanks for that check; I didn't realize SDL_Event was [a union with padding and such](https://github.com/icebreaker/sdl2/blob/39440e1201fcb00cd9d145842a77d7bbcc13f5e9/include/SDL_events.h#L495). Interesting!

(Yeah, just in general, using references to compound objects is a decent rule or thumb.)

Reply:
thanks for the C++ tutorial - changes rebased & amended!